### PR TITLE
June18 dev spectator view multilens support

### PR DESCRIPTION
**Pull request by JoseSantano-ms**
### Overview 
SpectatorView is now able to work with an existing multilens session **if such session uses UNET**. A new script has been created `SpectatorView.cs` that has a reference to a NetworkManager. From now on SpectatorView will only work if there's a NetworkManager in the scene, it's been referenced and the session started.

### Changes 
- SpectatorView can now use an external NetworkManager
- Changes in the error handling for the dependencies
- SpectatorView is now a single prefab
- Example scene changed to have an external NetworkManager

### CODE REVIEW

File: Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{
+    /// <summary>
+    /// Is the device a host or a client? (HoloLens or mobile?)
+    /// </summary>
+    private bool isHost;
+
+    // Use this for initialization
+    private void Start()
+    {
+        isHost = FindObjectOfType<PlatformSwitcher>().TargetPlatform == PlatformSwitcher.Platform.Hololens;
```
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs]: What is this PlatformSwitcher and why can't we either cache a reference to it or make it a singleton?

FindObjectOfType is very expensive.

Comment (JoseSantano-ms) [reply Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs]: Actually, let's change that to be consistent with the rest of the changes and ask the `SpectatorView` for the `isHost` field. Cached reference of course 👍

File: Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿using HoloToolkit.Unity.Preview.SpectatorView;
```
Comment (timGerken) [inline Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs]: @davidkline-ms Does sample code require the MIT license header?

File: Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+using UnityEngine;
+using UnityEngine.Networking;
+
+/// <summary>
+/// Example NetworkManager implementation. It simply starts tge server if the HoloLens is running the app.
```
Comment (timGerken) [inline Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs]: tge -> the

File: Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs (hunk 2 above)
Comment (JoseSantano-ms) [reply Assets/HoloToolkit-Examples/SpectatorView/Scripts/SpectatorViewNetworkManager.cs]: For consistency with ColorChanger.cs: Adding license header.

Thanks Tim!

**Pull request merged**